pkg/util/util.go

```diff
@@ -69,3 +69,21 @@ func ResourceNameToEnvVar(prefix string, resourceName string) string {
 	varName = strings.Replace(varName, ".", "_", -1)
 	return fmt.Sprintf("%s_%s", prefix, varName)
 }
+
+// Checks if kernel boot is defined in a valid way
+func IsKernelBootDefinedProperly(vmi *v1.VirtualMachineInstance) bool {
+	if vmi == nil {
+		return false
+	}
+
+	vmiFirmware := vmi.Spec.Domain.Firmware
+	if (vmiFirmware == nil) || (vmiFirmware.KernelBoot == nil) || (vmiFirmware.KernelBoot.Container == nil) {
+		return false
+	}
+
+	if c := vmiFirmware.KernelBoot.Container; (c.KernelPath == "") && (c.InitrdPath == "") {
```

Comment:
I think it should be fine to specify initrd without the kernel image - I don't see that libvirt nor qemu prevents this.

Reply:
You're absolutely right.
This condition is true (which means that `false` will be returned == kernel boot is not defined properly) only if both kernel and initid are not defined.